tests/ui/type-alias-impl-trait/non-lifetime-binder.rs

```diff
@@ -5,6 +5,7 @@ trait Trait<T> {}
 
 fn f() -> impl for<T> Trait<impl Trait<T>> {}
 //~^ ERROR nested `impl Trait` is not allowed
-//~| ERROR the trait bound `(): Trait<impl Trait<T>>` is not satisfied
+//~| ERROR the trait bound `(): Trait<impl Trait<{type error}>>` is not satisfied
```

Comment:
This error message sucks, but also it's the same error message we'd get from something like `impl Trait<impl Trait<Missing>>`, and really has nothing to do with this test other than now we begin resolving the late-bound `T` to error since it crosses a late-bound scope. 

The predicate itself doesn't reference errors, but when we render an opaque using its bounds, we'll render `{type error}`. We could suppress it, but also I don't think it needs to be fixed here.